src/client/sky.cpp

```diff
@@ -640,113 +620,218 @@ void Sky::draw_sun(video::IVideoDriver *driver, float sunsize, const video::SCol
+				vertices[i * 3 + 2].Pos = p2;
+				
+
+				vertices[i * 3 + 2].Color = starcolor;
+			}
+			driver->drawIndexedTriangleList(vertices.data(), m_star_count * 3,
+					indices.data(), m_star_count);
+			delete[] indices;
+			delete[] vertices;
+#else
+			u16 *indices = new u16[m_star_count * 4];
+			video::S3DVertex *vertices =
+					new video::S3DVertex[m_star_count * 4];
+			for (u32 i = 0; i < m_star_count; i++) {
+				indices[i * 4 + 0] = i * 4 + 0;
+				indices[i * 4 + 1] = i * 4 + 1;
+				indices[i * 4 + 2] = i * 4 + 2;
+				indices[i * 4 + 3] = i * 4 + 3;
+				v3f r = m_stars[i];
+				core::CMatrix4<f32> a;
+				a.buildRotateFromTo(v3f(0, 1, 0), r);
+				v3f p = v3f(-d, 1, -d);
+				v3f p1 = v3f(d, 1, -d);
+				v3f p2 = v3f(d, 1, d);
+				v3f p3 = v3f(-d, 1, d);
+				a.rotateVect(p);
+				a.rotateVect(p1);
+				a.rotateVect(p2);
+				a.rotateVect(p3);
+				p.rotateXYBy(wicked_time_of_day * 360 - 90);
+				p.rotateXZBy(m_star_rotation);
+				p1.rotateXYBy(wicked_time_of_day * 360 - 90);
+				p1.rotateXZBy(m_star_rotation);
+				p2.rotateXYBy(wicked_time_of_day * 360 - 90);
+				p2.rotateXZBy(m_star_rotation);
+				p3.rotateXYBy(wicked_time_of_day * 360 - 90);
+				p3.rotateXZBy(m_star_rotation);
+				vertices[i * 4 + 0].Pos = p;
+				vertices[i * 4 + 0].Color = starcolor;
+				vertices[i * 4 + 1].Pos = p1;
+				vertices[i * 4 + 1].Color = starcolor;
+				vertices[i * 4 + 2].Pos = p2;
+				vertices[i * 4 + 2].Color = starcolor;
+				vertices[i * 4 + 3].Pos = p3;
+				vertices[i * 4 + 3].Color = starcolor;
+			}
+			driver->drawVertexPrimitiveList(vertices, m_star_count * 4,
+					indices, m_star_count, video::EVT_STANDARD,
+					scene::EPT_QUADS, video::EIT_16BIT);
+			delete[] indices;
+			delete[] vertices;
+#endif
+	}
 
-void Sky::draw_sky_body(std::array<video::S3DVertex, 4> &vertices, float pos_1, float pos_2, const video::SColor &c)
-{
-	/*
-	 * Create an array of vertices with the dimensions specified.
-	 * pos_1, pos_2: position of the body's vertices
-	 * c: color of the body
-	 */
+	void Sky::draw_sky_body(std::array<video::S3DVertex, 4> &vertices, float pos_1, float pos_2, const video::SColor &c)
+	{
+		/*
+		 * Create an array of vertices with the dimensions specified.
+		 * pos_1, pos_2: position of the body's vertices
+		 * c: color of the body
+		 */
 
-	const f32 t = 1.0f;
-	const f32 o = 0.0f;
-	vertices[0] = video::S3DVertex(pos_1, pos_1, -1, 0, 0, 1, c, t, t);
-	vertices[1] = video::S3DVertex(pos_2, pos_1, -1, 0, 0, 1, c, o, t);
-	vertices[2] = video::S3DVertex(pos_2, pos_2, -1, 0, 0, 1, c, o, o);
-	vertices[3] = video::S3DVertex(pos_1, pos_2, -1, 0, 0, 1, c, t, o);
-}
+		const f32 t = 1.0f;
+		const f32 o = 0.0f;
+		vertices[0] = video::S3DVertex(pos_1, pos_1, -1, 0, 0, 1, c, t, t);
+		vertices[1] = video::S3DVertex(pos_2, pos_1, -1, 0, 0, 1, c, o, t);
+		vertices[2] = video::S3DVertex(pos_2, pos_2, -1, 0, 0, 1, c, o, o);
+		vertices[3] = video::S3DVertex(pos_1, pos_2, -1, 0, 0, 1, c, t, o);
+	}
```

Comment:
indented for no reason

Reply:
Not resolved, but not an essential change.

Reply:
Should it stay indented or revert to as it was?

Reply:
Yes, please do fix obvious code style mistakes.